txnkv/transaction/txn.go

```diff
@@ -1140,7 +1140,10 @@ func (txn *KVTxn) lockKeys(ctx context.Context, lockCtx *tikv.LockCtx, fn func()
 				}
 			}
 		}
-		if txn.aggressiveLockingContext != nil {
+
+		// note that lock_only_if_exists guarantees the response tells us whether the value exists
+		keyIsNotLocked := lockCtx.LockOnlyIfExists && !valExists
+		if txn.IsInAggressiveLockingMode() && !keyIsNotLocked {
```

Comment:
For
```
if txn.IsInAggressiveLockingMode() && keyIsNotLocked
```
branch, it seems the `aggressiveLockingContext` is not updated, is it unexpected?

Reply:
Sorry I don't get it. `txn.aggressiveLockingContext.currentLockedKeys` is updated in this branch.